app/assets/javascripts/spree/frontend/spree_payu_integration.js

```diff
@@ -1 +1,31 @@
 /* Placeholder for backend dummy app */
+SpreePayuIntegration = {
+  updateSaveAndContinueVisibility: function() {
+    if (this.isButtonHidden()) {
+      $(this).trigger('hideSaveAndContinue')
+    } else {
+      $(this).trigger('showSaveAndContinue')
+    }
+  },
+  isButtonHidden: function () {
+    paymentMethod = this.checkedPaymentMethod();
+    return (SpreePayuIntegration.paymentMethodID && paymentMethod.val() == SpreePayuIntegration.paymentMethodID);
+  },
+  checkedPaymentMethod: function() {
+    return $('div[data-hook="checkout_payment_step"] input[type="radio"][name="order[payments_attributes][][payment_method_id]"]:checked');
```

Comment:
This scares me like a shit and I do not understand this chain. Why not passing the selected value from the checkbox which is clicked/changed and you already have it in click event in line 27?